documentation/public/user-guide/concepts.md

```diff
@@ -1,11 +1,57 @@
-At its core, Slang is a collection of APIs that are meant to analyze the source code, starting with the source code itself and ending with a rich structure that can be reasoned about. This is a departure from the classic approach of "black-box" compilers, which are handed the input and only their output can be observed.
+# Concepts
+
+At its core, Slang is a collection of APIs that are meant to analyze the source code, starting with the source code itself and ending with a rich structure that can be reasoned about.
+This is a departure from the classic approach of "black-box" compilers, which are handed the input and only their output can be observed.
+
+## Language Versions
+
+To use Slang, you start by initializing a `Language` object with a specific version of the language.
+The earliest Solidity version we support is `0.4.11`, and we plan on supporting all future versions as they are released.
+
+From a `Language` object, you can analyze any source text according to the rules of that specific version.
+Providing an accurate language version is important, as it affects the shape of the syntax tree, and possible errors produced.
+You can use the `Language::getSupportedVersions()` API to get a list of all supported versions for the current Slang release.
+
+The `Language::parse()` API is the main entry point for the parser, and to generate concrete syntax trees (CSTs) that can be used for further analysis.
+Each `parse()` operation accepts the input source code, and a `RuleKind` variant.
+This allows callers to parse entire source files (`RuleKind::SourceUnit`), individual contracts (`RuleKind::ContractDefinition`),
+methods (`RuleKind::FunctionDefinition`), or any other syntax nodes.
```

Comment:
I am still not sold on this, given how much there's code/blocks here and in the CST Cursors section and supporting multiple versions as an idea/concept is explained already in User Guide > Introduction.

I don't think this is also a good place to list all different kinds of node relationships the cursor can jump to (to give an example what I mean by a lot of code), but given how this PR improves multiple other different things, I don't want to block on this and we can try to make it clearer in a follow-up.